physical/gcs/gcs_ha.go

```diff
@@ -180,10 +180,30 @@ func (l *Lock) Unlock() error {
 	}
 	l.stopLock.Unlock()
 
-	// Delete
+	// Read the record value before deleting. This needs to be a CAS operation or
+	// else we might be deleting someone else's lock.
 	ctx := context.Background()
-	if err := l.backend.Delete(ctx, l.key); err != nil {
-		return err
+	r, err := l.get(ctx)
+	if err != nil {
+		return errwrap.Wrapf("failed to read lock for deletion: {{err}}", err)
+	}
+	if r != nil && r.Identity == l.identity {
+		ctx := context.Background()
```

Comment:
This ctx call is unnecessary, ctx from above is already Background.

Reply:
I've been bitten in the past during refactoring where it's unclear that a context is being used further down the method, so I'd prefer to keep the call to .get and the call to .Delete to use separate contexts that are explicit.